beacon-chain/state/stategen/epoch_boundary_state_cache.go

```diff
@@ -67,8 +67,8 @@ func newBoundaryStateCache() *epochBoundaryState {
 }
 
 // ByRoot satisfies the CachedGetter interface
-func (e *epochBoundaryState) ByRoot(r [32]byte) (state.BeaconState, error) {
-	rsi, ok, err := e.getByRoot(r)
+func (e *epochBoundaryState) ByRoot(blockRoot [32]byte) (state.BeaconState, error) {
+	rsi, ok, err := e.getByRoot(blockRoot)
```

Comment:
Go style generally prefers smaller variable names. https://github.com/golang/go/wiki/CodeReviewComments#variable-names

In this case, I would recommend leaving the root as `r` because this method is aptly named and very short. It is clear to everyone that `r` is a root, although it is not clear what kind of root it is. 

Perhaps there is an argument that this method is not aptly named since there is ambiguity around what kind of root it is. My suggestion is to change this to `ByBlockRoot(br [32]byte)` then we all know what we are dealing with.